### PR TITLE
[#110] [Integrate] As a user I can see zoom in and zoom out animation in survey detail screen

### LINKED ISSUE #110: [Integrate] As a user I can see zoom in and zoom out animation in survey detail screen
## Why

According to Figma, navigating from survey screen to detail screen will zoom in the background image. When the user exit from the detail screen, background image will zoom out.

## Acceptance Criteria

- Zoom in the background image when navigating to survey detail screen
- Zoom out and return to original size when navigate back from survey detail screen

## Resource
https://www.figma.com/proto/GjRPOjDyZ6f4EDL3wKarRK/Challenge---Mobile-App?node-id=44%3A148&scaling=scale-down&page-id=0%3A1&starting-point-node-id=34%3A450

### PR DESCRIPTION
**Pull request by kaungkhantsoe**
Closes #110 

## What happened 👀

According to Figma, navigating from survey screen to detail screen will zoom in the background image. When the user exit from the detail screen, background image will zoom out.

## Insight 📝

Tried zooming with `1f` and also `0.5f` per `25 milli seconds` but the animation seems to be lagging. So I tried using `0.025f` per `25 milli seconds` and animation becomes smoother.

## Proof Of Work 📹

https://user-images.githubusercontent.com/32578035/196386972-48d74194-f14f-439c-97c9-89810f8906a3.mp4

### CODE REVIEW

**Comment by ryan-conway** [comment]
@kaungkhantsoe any idea how we can solve the white flash when transitioning? What if we were to use a dark theme instead to have the background as black instead of white? 🤔

File: app/src/main/java/com/kks/nimblesurveyjetpackcompose/model/SurveyQuestion.kt [hunk 1]
```diff
@@ -7,19 +7,24 @@ data class SurveyQuestion(
...
 enum class QuestionDisplayType(val typeValue: String, val isIdOnlyAnswer: Boolean) {
     NONE("none", false),
     INTRO("intro", false),
     DROPDOWN("dropdown", true),
     SMILEY("smiley", true),
     THUMBS("thumbs", true),
     STARS("star", true),
-    NPS("nps", true)
+    NPS("nps", true),
+    CHOICE("choice",true)
```
Comment (AVI5HEK) [inline app/src/main/java/com/kks/nimblesurveyjetpackcompose/model/SurveyQuestion.kt]:
```suggestion
    CHOICE("choice", true)
```

File: app/src/main/java/com/kks/nimblesurveyjetpackcompose/model/response/IncludedResponse.kt [hunk 2]
```diff
@@ -48,11 +49,20 @@ fun IncludedQuestionResponse.toSurveyQuestion() =
...
         answers = emptyList()
     )
 
+private const val SINGLE_PICK = "one"
+
+fun String?.toSurveyQuestionPickType(): SurveyQuestionPickType {
+    this?.let { pick ->
+        return if (pick == SINGLE_PICK) SurveyQuestionPickType.SINGLE else SurveyQuestionPickType.MULTIPLE
+    }
+    return SurveyQuestionPickType.NONE
```
Comment (AVI5HEK) [inline app/src/main/java/com/kks/nimblesurveyjetpackcompose/model/response/IncludedResponse.kt]: Optional:
```suggestion
    return this?.let { pick ->
        if (pick == SINGLE_PICK) SurveyQuestionPickType.SINGLE else SurveyQuestionPickType.MULTIPLE
    } ?: SurveyQuestionPickType.NONE
```

**Review by AVI5HEK — COMMENTED** [review]
Overall lgtm. Need to resolve the conflicts.

**Comment by kaungkhantsoe** [comment]
> @kaungkhantsoe any idea how we can solve the white flash when transitioning? What if we were to use a dark theme instead to have the background as black instead of white? 🤔

@ryan-conway Updated at 55cb9a5aeb6439bc6b4c1f753127270286a90754 . Also updated the [POW](https://github.com/kaungkhantsoe/NimbleSurveyJetpackCompose/pull/111#issue-1408848650)

**Review by AVI5HEK — APPROVED** [review]
LGTM 🚀

**Review by ryan-conway — APPROVED** [review]
LGTM 🙏🏻

**Pull request merged**